pkg/sql/sem/builtins/builtins.go

```diff
@@ -6384,6 +6384,9 @@ table's zone configuration this will return NULL.`,
 						"can only deserialize matching session users",
 					)
 				}
+				if err := evalCtx.Planner.CheckCanBecomeUser(evalCtx.Context, sd.User()); err != nil {
```

Comment:
actually, why don't we just make it so that `sd.CurrentUser().Normalized()` must match `evalCtx.SessionData().CurrentUser().Normalized()`?

it's a simpler check and we don't have any use case that needs to allow anyone to automatically change roles using this.

Reply:
because during a session migration someone might use `SET ROLE`, e.g. using postgrest.

i don't think we should error that out

Reply:
ah good call. i should also update `create_session_revival_token` so it uses the sessionUser rather than currentUser for the token then.